### PR TITLE
[Update] MSstatsConverter (TMT)

### PR DESCRIPTION
**Pull request by oliveralka**
Some minor updates for the MSstatsConverter, due to changes in the next R package version (MSstatsTMT). 

Question: 
The class MSstatsFile, due to the addition of the TMT-method has a lot of code duplication, or at least code overlap, with small changes e.g. MSstatsFile::MSstatsLine and MSstatsFile::MSstatsTMTLine; storeLFQ and storeISO.

Here, I tried to do some refactoring, but I am a little bit lost - where do I start and what would be best to make the code a bit cleaner and more functional. 

I would be very happy if you could give me some pointers! 
This will then probably go to a new PR.

### CODE REVIEW

File: src/openms/include/OpenMS/FORMAT/MSstatsFile.h [hunk 1]
```diff
@@ -63,6 +63,21 @@ namespace OpenMS
...
+        struct AggregatedConsensusInfo
+        {
+          std::vector< std::vector< String > > consensus_feature_filenames;           // Filenames of ConsensusFeature
+          std::vector< std::vector< Intensity > > consensus_feature_intensites;       // Intensites of ConsensusFeature
+          std::vector< std::vector< Coordinate > > consensus_feature_retention_times; // Retention times of ConsensusFeature
+          std::vector< std::vector< unsigned > > consensus_feature_labels;          // Labels of ConsensusFeature
+        };
+
+        MSstatsFile::AggregatedConsensusInfo aggregateInfo(ConsensusMap const &consensus_map,
+                                                           std::vector<BaseFeature> &features,
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: I find it a bit confusing that there are two "results" of this function.
The column vectors (=AggregatedConsensusInfo) and &features

File: src/openms/source/FORMAT/MSstatsFile.cpp [hunk 2]
```diff
@@ -569,13 +539,13 @@ void OpenMS::MSstatsFile::storeISO(const OpenMS::String &filename, const Consens
           for (const OpenMS::PeptideEvidence &pep_ev : peptide_evidences)
           {
             // Write new line for each run
-            for (Size j = 0; j < consensus_feature_filenames[i].size(); j++)
+            for (Size j = 0; j < AggregatedInfo.consensus_feature_filenames[i].size(); j++)
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/MSstatsFile.cpp]: I find it a bit confusing that you are iterating over the features in the outer loop but also take the information from the AggregatedInfo datastructure.

File: src/openms/include/OpenMS/FORMAT/MSstatsFile.h (hunk 1 above)
Comment (timosachsenberg) [reply src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: could this function and struct be private?

Comment (oliveralka) [reply src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: The function could and struct could be private. 

And you are right - there should not be two "results". I was not paying enough attention to that while extracting the method. I actually wanted to store all relevant information in AggregatedConsensusInfo.

File: src/openms/source/FORMAT/MSstatsFile.cpp (hunk 2 above)
Comment (oliveralka) [reply src/openms/source/FORMAT/MSstatsFile.cpp]: yes - see comment above.

File: src/openms/include/OpenMS/FORMAT/MSstatsFile.h [hunk 3]
```diff
@@ -82,14 +82,34 @@ namespace OpenMS
...
+        typedef OpenMS::Peak2D::CoordinateType Coordinate;
+
         const String na_string = "NA";
 
+
         /*
-        *  @brief: Internal function to check if MSstats_BioReplicate and MSstats_Condition in Experimental Design
-        */
+         *  @brief: Struct to aggregate intermediate information from ConsensusFeature and ConsensusMap,
+         *  such as filenames, intensities, retention times, labes and features (for further processing)
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: typo: labels

File: src/openms/include/OpenMS/FORMAT/MSstatsFile.h [hunk 4]
```diff
@@ -82,14 +82,34 @@ namespace OpenMS
...
+         */
+        struct AggregatedConsensusInfo
+        {
+          std::vector< std::vector< String > > consensus_feature_filenames;           // Filenames of ConsensusFeature
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: use proper doxygen commnts //<

Comment (timosachsenberg) [reply src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: add information what the entries correspond to: e.g. consensus_feature_filenames[0] corresponds to the the filenames stored in the first ConsensusMap

File: src/openms/include/OpenMS/FORMAT/MSstatsFile.h [hunk 5]
```diff
@@ -82,14 +82,34 @@ namespace OpenMS
...
+          std::vector< std::vector< Intensity > > consensus_feature_intensities;       // Intensities of ConsensusFeature
+          std::vector< std::vector< Coordinate > > consensus_feature_retention_times; // Retention times of ConsensusFeature
+          std::vector< std::vector< unsigned > > consensus_feature_labels;          // Labels of ConsensusFeature
+          std::vector<BaseFeature> features;                                        // Features of ConsensusMap
+        };
+
+        MSstatsFile::AggregatedConsensusInfo aggregateInfo(const ConsensusMap& consensus_map,
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: docuement or make private

File: src/openms/include/OpenMS/FORMAT/MSstatsFile.h [hunk 6]
```diff
@@ -82,14 +82,34 @@ namespace OpenMS
...
         const String na_string = "NA";
 
+
         /*
-        *  @brief: Internal function to check if MSstats_BioReplicate and MSstats_Condition in Experimental Design
-        */
+         *  @brief: Struct to aggregate intermediate information from ConsensusFeature and ConsensusMap,
+         *  such as filenames, intensities, retention times, labes and features (for further processing)
+         */
+        struct AggregatedConsensusInfo
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: can this struct be private? AggregatedConsensusInfo_

File: src/openms/include/OpenMS/FORMAT/MSstatsFile.h [hunk 7]
```diff
@@ -82,14 +82,34 @@ namespace OpenMS
...
+          std::vector< std::vector< Coordinate > > consensus_feature_retention_times; // Retention times of ConsensusFeature
+          std::vector< std::vector< unsigned > > consensus_feature_labels;          // Labels of ConsensusFeature
+          std::vector<BaseFeature> features;                                        // Features of ConsensusMap
+        };
+
+        MSstatsFile::AggregatedConsensusInfo aggregateInfo(const ConsensusMap& consensus_map,
+                                                           const std::vector<String>& spectra_paths);
+
+        /*
+         *  @brief: Internal function to check if MSstats_BioReplicate and MSstats_Condition in Experimental Design
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: *exists* in

File: src/openms/include/OpenMS/FORMAT/MSstatsFile.h [hunk 8]
```diff
@@ -144,142 +163,154 @@ namespace OpenMS
...
+        String fragment_ion_;
+        String frag_charge_;
+        String isotope_label_type_;
+        String condition_;
+        String bioreplicate_;
+        String run_;
+        String fraction_;
+      };
+
         class MSstatsTMTLine
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: different intendation? see line below

File: src/openms/include/OpenMS/FORMAT/MSstatsFile.h [hunk 9]
```diff
@@ -144,142 +163,154 @@ namespace OpenMS
...
+        String mixture_;
+        String techrepmixture_;
+        String fraction_;
+      };
+
+        /*
+         *  @brief Constructs the lines and adds them to the TextFile
+         */
+        template <class LineTemplate>
+        void constructFile(const String& retention_time_summarization_method,
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/FORMAT/MSstatsFile.h]: private? _

**Pull request merged**